manim/mobject/geometry/labeled.py

```diff
@@ -2,65 +2,148 @@
 
 from __future__ import annotations
 
-__all__ = ["LabeledLine", "LabeledArrow"]
+__all__ = ["Label", "LabeledLine", "LabeledArrow", "LabeledPolygram"]
 
 from typing import TYPE_CHECKING
 
+import numpy as np
+
 from manim.constants import *
 from manim.mobject.geometry.line import Arrow, Line
+from manim.mobject.geometry.polygram import Polygram
 from manim.mobject.geometry.shape_matchers import (
     BackgroundRectangle,
     SurroundingRectangle,
 )
 from manim.mobject.text.tex_mobject import MathTex, Tex
 from manim.mobject.text.text_mobject import Text
-from manim.utils.color import WHITE, ManimColor, ParsableManimColor
+from manim.mobject.types.vectorized_mobject import VGroup
+from manim.utils.color import WHITE
+from manim.utils.polylabel import polylabel
 
 if TYPE_CHECKING:
     from typing import Any
 
+    from manim.typing import Point3D
+
+
+class Label(VGroup):
+    """A Label consisting of text surrounded by a frame.
+
+    Parameters
+    ----------
+    label
+        Label that will be displayed.
+    label_config
+        A dictionary containing the configuration for the label.
+        This is only applied if `label` is of type `str`.
+    box_config
+        A dictionary containing the configuration for the background box.
+    frame_config
+         A dictionary containing the configuration for the frame.
+
+    Examples
+    --------
+    .. manim:: LabelExample
+        :save_last_frame:
+        :quality: high
+
+        class LabelExample(Scene):
+            def construct(self):
+                label = Label(
+                    label=Text('Label Text', font='sans-serif'),
+                    box_config = {
+                        "color" : BLUE,
+                        "fill_opacity" : 0.75
+                    }
+                )
+                label.scale(3)
+                self.add(label)
+    """
+
+    def __init__(
+        self,
+        label: str | Tex | MathTex | Text,
+        label_config: dict | None = None,
+        box_config: dict | None = None,
+        frame_config: dict | None = None,
```

Comment:
Using `dict[str, Any]` instead of just `dict` is preferable here for two reasons:
- The keys of `dict` can be any immutable object, but here we require that these config dictionaries have `str` keys.
- Even if we had `dict[Any, Any]`, we would prefer to be explicit anyways.

```suggestion
        label_config: dict[str, Any] | None = None,
        box_config: dict[str, Any] | None = None,
        frame_config: dict[str, Any] | None = None,
```